src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/problem/ExceptionHandling.kt

```diff
@@ -130,6 +135,19 @@ class ExceptionHandling(
     return true
   }
 
+  override fun log(throwable: Throwable, problem: Problem, request: NativeWebRequest, status: HttpStatus) {
+    when (problem) {
+      is BadRequestProblem -> logBadRequestProblem(problem)
+      is InternalServerErrorProblem -> logInternalServerErrorProblem(problem)
```

Comment:
Should we be calling `logInternalServerErrorProblem` somewhere too? (like we do with `logBadRequestProblem` on line 115) Or is there some magic going on here?

Reply:
this `log` method is from the `ProblemHandling` interface and gets called whenever we throw a Problem from a controller.  We only call logBadRequestProblem explicitly from line 115 because this `log` function never gets called on that particular route since it never reaches a controller.

Reply:
Ah, gotcha